File: Java/Aksesoris.java
class Aksesoris extends PetShop {
    private String jenis;
    private String bahan;
    private String warna;

    public Aksesoris() {
        this.jenis = "";
        this.bahan = "";
        this.warna = "";
    }

    public Aksesoris(int id, String namaProduk, int hargaProduk, int stokProduk, String jenis, String bahan, String warna) {
        super(id, namaProduk, hargaProduk, stokProduk);
        this.jenis = jenis;
        this.bahan = bahan;
        this.warna = warna;
    }

    public void setJenis(String jenis) {
        this.jenis = jenis;
    }

    public String getJenis() {
        return jenis;
    }

    public void setBahan(String bahan) {
        this.bahan = bahan;
    }

    public String getBahan() {
        return bahan;
    }

    public void setWarna(String warna) {
        this.warna = warna;
    }

    public String getWarna() {
        return warna;
    }

    public void tampilkanData(int panjangId, int panjangNama, int panjangHarga, int panjangStok, int panjangJenis, int panjangBahan, int panjangWarna) {
        super.tampilkanData(panjangId, panjangNama, panjangHarga, panjangStok);
        System.out.printf("%-" + panjangJenis + "s%-" + panjangBahan + "s%-" + panjangWarna + "s", jenis, bahan, warna);
    }
}

File: CPP/Aksesoris.cpp
#include "Petshop.cpp"

using namespace std;

class Aksesoris : public PetShop {
private:
    string jenis;
    string bahan;
    string warna;

public:
    Aksesoris() {
        this->jenis = "";
        this->bahan = "";
        this->warna = "";
    }

    Aksesoris(int id, string namaProduk, int hargaProduk, int stokProduk, string jenis, string bahan, string warna) 
        : PetShop(id, namaProduk, hargaProduk, stokProduk), jenis(jenis), bahan(bahan), warna(warna) {
            this->jenis = jenis;
            this->bahan = bahan;
            this->warna = warna;
        }

    void setJenis(const string &jenis) {this->jenis = jenis;}
    string getJenis() const {return jenis;}

    void setBahan(const string &bahan) {this->bahan = bahan;}
    string getBahan() const {return bahan;}

    void setWarna(const string &warna) {this->warna = warna;}
    string getWarna() const {return warna;}

    void tampilkanData(int panjang_id, int panjang_nama, int panjang_harga, int panjang_stok, int panjang_jenis, int panjang_bahan, int panjang_warna) const {
        PetShop::tampilkanData(panjang_id, panjang_nama, panjang_harga, panjang_stok);
        cout << setw(panjang_jenis) << jenis
             << setw(panjang_bahan) << bahan
             << setw(panjang_warna) << warna;
    }

    ~Aksesoris() {}
};